server/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -93,15 +94,52 @@
                 }
             };
 
-    public static final Setting<Integer> SCRIPT_CACHE_SIZE_SETTING =
+    public static final Setting<Integer> SCRIPT_CACHE_SIZE_SETTING_DEPRECATED =
```

Comment:
I don't think we've used this naming pattern of suffixing with _DEPRECATED before. If we are going to change the object name, perhaps instead we could refer to this as the "general" version since that is the terminology you used elsewhere? eg SCRIPT_GENERAL_CACHE_SIZE_SETTING. We also need the Property.Deprecated on these.

Reply:
> We also need the Property.Deprecated on these.

That will be added in a future change, see [step 5](https://github.com/elastic/elasticsearch/issues/50152#issuecomment-587927041)

Reply:
Changed to `SCRIPT_GENERAL_`, leaving the deprecated property addition to a future change.

Reply:
As mentioned, deprecation will happen in a future change.  There's a bunch of tests, etc that need to change in tandem with that.